tests/test_middleware_interface.py

```diff
@@ -219,7 +228,17 @@ def test_ingest_image_fails_missing_file(self):
         self.assertEqual(datasets, [])
 
     def test_run_pipeline(self):
-        with self.assertLogs(self.logger_name, level="INFO") as cm:
-            self.interface.run_pipeline(self.next_visit, 1)
-        msg = f"INFO:{self.logger_name}:Running pipeline bias.yaml on visit '{self.next_visit}', snaps 1"
-        self.assertEqual(cm.output, [msg])
+        # Can't run an actual pipeline because raw/calib/refcat/template data are all zeroed out.
+        with unittest.mock.patch('activator.middleware_interface.SimplePipelineExecutor') as mock_executor:
+            self.interface.run_pipeline(self.next_visit, {1})
+            mock_executor.from_pipeline.assert_called_once_with(
+                self.interface.pipeline,
+                where=f"detector={self.next_visit.detector} and exposure in (1)",
+                butler=self.interface.butler
+            )
+            pipeline = mock_executor.from_pipeline(
```

Comment:
These are some unfortunately chosen variable names, in hindsight. I'd suggest calling `mock_executor` either `mock_class` or `mock_executor_class`, and renaming `pipeline` to `mock_executor` or `mock_executor_instance`.